File: app/src/main/java/com/musicplayer/db/entity/PlaylistSong.java
package com.musicplayer.db.entity;

import androidx.annotation.NonNull;
import androidx.room.Entity;

@Entity(tableName = "playlist_songs", primaryKeys = {"playlistId", "songId"})
public class PlaylistSong {
    @NonNull
    public int playlistId; // Aunque int no es nulo, Room lo pide a veces para claves primarias compuestas
    @NonNull
    public String songId;
    public String title;
    public String artist;
    public String thumbnailUrl;
    public String url;
    public boolean isLocal;

    public PlaylistSong(int playlistId, String songId, String title, String artist, String thumbnailUrl, String url, boolean isLocal) {
        this.playlistId = playlistId;
        this.songId = songId;
        this.title = title;
        this.artist = artist;
        this.thumbnailUrl = thumbnailUrl;
        this.url = url;
        this.isLocal = isLocal;
    }
}

File: app/src/main/java/com/musicplayer/db/entity/PlaylistSong.kt
package com.musicplayer.db.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index
import kotlin.jvm.JvmField

@Entity(
    tableName = "playlist_songs",
    primaryKeys = ["playlistId", "songId"],
    foreignKeys = [ForeignKey(
        entity = Playlist::class,
        parentColumns = ["id"],
        childColumns = ["playlistId"],
        onDelete = ForeignKey.CASCADE
    )],
    indices = [Index("playlistId"), Index("songId")]
)
data class PlaylistSong(
    @JvmField var playlistId: Int = 0,
    @JvmField var songId: String = "",
    @JvmField var title: String? = null,
    @JvmField var artist: String? = null,
    @JvmField var thumbnailUrl: String? = null,
    @JvmField var url: String? = null,
    @JvmField var isLocal: Boolean = false,
    @JvmField var cachedStreamUrl: String? = null,
    @JvmField var cachedAt: Long = 0
)
